File: el megacodigo.cpp
#include <iostream>

/* run this program using the console pauser or add your own getch, system("pause") or input loop */
using namespace std;
int main(int argc, char** argv) {
	int a,n,m,x,exp;
	cout<<"base: ";
	cin>>a;
	cout<<"exponente: ";
	cin>>n;
	cout<<"modulo: ";
	cin>>m;
	exp=1;
	x=a%m;
	while (n>0){
		if (n%2!=0){
			exp=(exp*x)%m;
		}
		x=(x*x)%m;
		n=n/2;
	}
	cout<<exp;
	return 0;
}

File: el megacodigo.py
a=int(input("base: "))
n=int(input("exponente: "))
m=int(input("modulo: "))
decimal=n

def binario(decimal):
    bin = ''
    while decimal // 2 != 0:
        bin = str(decimal % 2) + bin
        decimal = decimal // 2
    return str(decimal) + bin
print("Exponente en binario: ", binario(n))

res=1
b=a % m
while (n>0):
    if (n%2!=0):
        res=(res*b)%m
    b=(b*b)%m
    n=n//2
print("El resto es: ",res)